pkg/odo/util/pushtarget/pushtarget.go

```diff
@@ -0,0 +1,45 @@
+package pushtarget
+
+import (
+	"os"
+
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/preference"
+)
+
+// env variables
+const (
+	// Setting this env to `docker` will enable pushing to docker containers
+	// and will override the setting in the preferences file.
+
+	OdoPushTarget = "ODO_PUSHTARGET"
+)
+
+// IsPushTargetDocker checks if the push target preference has been set to docker
+func IsPushTargetDocker() bool {
+
+	// ODO's push target can be told to use docker by:
+	//		- setting an env variable "ODO_PUSHTARGET=docker" or
+	//		- setting odo preference using "odo preference set pushtarget docker"
+
+	var (
+		pushTargetEnv        string
+		pushTargetPreference string
+	)
+
+	// Check "ODO_PUSHTARGET" env variable.
+	pushTargetEnv, _ = os.LookupEnv(OdoPushTarget)
+	if pushTargetEnv == "docker" || pushTargetEnv == "kube" {
```

Comment:
If I set `ODO_PUSHTARGET=foo" than it will get silently ignored and value will be taken from the preference file. 
Wouldn't it be better to fail if there is a nonsense value?

Reply:
That seems like a reasonable approach 👍 , I've updated the code to log an error and erturb false if the environment var has a nonsense value (and included a new test to catch that)

Reply:
> Wouldn't it be better to fail if there is a nonsense value?

Or maybe continue and look into the preference file instead of failing?